runtime/compiler/aarch64/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -36,8 +40,1040 @@ void J9::ARM64::AheadOfTimeCompile::processRelocations()
+      case TR_InlinedVirtualMethod:
+      case TR_InlinedInterfaceMethod:
+         {
+         guard = (TR_VirtualGuard *) relocation->getTargetAddress2();
+
+         // Setup flags field with type of method that needs to be validated at relocation time
+         if (guard->getSymbolReference()->getSymbol()->getMethodSymbol()->isStatic())
+            flags = inlinedMethodIsStatic;
+         if (guard->getSymbolReference()->getSymbol()->getMethodSymbol()->isSpecial())
+            flags = inlinedMethodIsSpecial;
+         if (guard->getSymbolReference()->getSymbol()->getMethodSymbol()->isVirtual())
+            flags = inlinedMethodIsVirtual;
+
+         TR_ASSERT((flags & RELOCATION_CROSS_PLATFORM_FLAGS_MASK) == 0,  "reloFlags bits overlap cross-platform flags bits\n");
+         *flagsCursor |= (flags & RELOCATION_RELOC_FLAGS_MASK);
+
+         int32_t inlinedSiteIndex = guard->getCurrentInlinedSiteIndex();
+         *(uintptrj_t *) cursor = (uintptrj_t) inlinedSiteIndex;
+         cursor += SIZEPOINTER;
+
+         *(uintptrj_t *) cursor = (uintptrj_t) guard->getSymbolReference()->getOwningMethod(comp)->constantPool(); // record constant pool
+         cursor += SIZEPOINTER;
+
+         if (relocation->getTargetKind() == TR_InlinedInterfaceMethodWithNopGuard ||
+             relocation->getTargetKind() == TR_InlinedInterfaceMethod ||
+             relocation->getTargetKind() == TR_InlinedAbstractMethodWithNopGuard)
+            {
+            TR_InlinedCallSite *inlinedCallSite = &comp->getInlinedCallSite(inlinedSiteIndex);
+            TR_AOTMethodInfo *aotMethodInfo = (TR_AOTMethodInfo *) inlinedCallSite->_methodInfo;
+            resolvedMethod = aotMethodInfo->resolvedMethod;
+            }
+         else
+            {
+            resolvedMethod = guard->getSymbolReference()->getSymbol()->getResolvedMethodSymbol()->getResolvedMethod();
+            }
+
+         if (comp->getOption(TR_UseSymbolValidationManager))
+            {
+            TR_OpaqueMethodBlock *method = resolvedMethod->getPersistentIdentifier();
+            TR_OpaqueClassBlock *thisClass = guard->getThisClass();
+            uint16_t methodID = symValManager->getIDFromSymbol(static_cast<void *>(method));
+            uint16_t receiverClassID = symValManager->getIDFromSymbol(static_cast<void *>(thisClass));
+
+            uintptrj_t data = 0;
+            data = (((uintptrj_t)receiverClassID << 16) | (uintptrj_t)methodID);
+            *(uintptrj_t*)cursor = data;
+            }
+         else
+            {
+            *(uintptrj_t*)cursor = (uintptrj_t)guard->getSymbolReference()->getCPIndex(); // record cpIndex
+            }
+         cursor += SIZEPOINTER;
+
+         TR_OpaqueClassBlock *inlinedMethodClass = resolvedMethod->containingClass();
+         void *romClass = (void *) fej9->getPersistentClassPointerFromClassPointer(inlinedMethodClass);
+         uintptr_t romClassOffsetInSharedCache = sharedCache->offsetInSharedCacheFromPointer(romClass);
```

Comment:
If https://github.com/eclipse/openj9/pull/7379 gets merged first, this will need to change to `self()->offsetInSharedCacheFromPointer(sharedCache, romClass);`

Reply:
Assuming #7379 gets merged first, I changed all the calls to `offsetInSharedCacheFromPointer` as proposed.